[polly] Port polly tests to use NPM

rahulana-quic opened the pull request:
Even as the NPM has been in use by Polly for a while now, the
majority of the tests continue using the LPM passes. This patch
ports the tests to use the NPM passes (for example, by replacing
a flag such as -polly-detect with -passes=polly-detect following
the NPM syntax for specifying passes) with some exceptions for
some missing features in the new passes. Additionally, the lit
substitution %loadPolly is replaced by the substitution of what
was %loadNPMPolly and %loadNPMPolly is removed.

rahulana-quic committed:
[polly] Port polly tests to use NPM

Even as the NPM has been in use by Polly for a while now, the
majority of the tests continue using the LPM passes. This patch
ports the tests to use the NPM passes (for example, by replacing
a flag such as -polly-detect with -passes=polly-detect following
the NPM syntax for specifying passes) with some exceptions for
some missing features in the new passes.

rahulana-quic commented:
Some tests such as the following have not been changed yet since some of the prints of the new passes (eg. JSONImportPass, ScopDetect) are not the same as the old passes.

  Polly :: CodeGen/MemAccess/create_arrays.ll
  Polly :: CodeGen/MemAccess/create_arrays_heap.ll
  Polly :: CodeGen/MemAccess/default_aligned_new_access_function.ll
  Polly :: CodeGen/MemAccess/map_scalar_access.ll
  Polly :: CodeGen/MemAccess/simple_analyze.ll
  Polly :: CodeGen/OpenMP/new_multidim_access.ll
  Polly :: DeLICM/reduction_constant_selfconflict.ll
  Polly :: ScopDetect/base_pointer.ll
  Polly :: ScopDetect/tlr_is_hoistable_load.ll
  Polly :: ScopInfo/aliasing_many_parameters_not_all_involved.ll
  Polly :: ScopInfo/bool-addrec.ll
  Polly :: ScopInfo/error-blocks-3.ll
  Polly :: ScopInfo/multidim_invalid_dimension.ll
  Polly :: ScopInfo/pr38218.ll
  Polly :: Simplify/redundant_partialwrite.ll

aeubanks reviewed (approved):
I obviously didn't look through every file, but seems right to me

please mention in the commit description that you've replaced the `%loadPolly` lit substitution with the previous `%loadNPMPolly` and removed `%loadNPMPolly`

rahulana-quic committed:
amend! [polly] Port polly tests to use NPM

Even as the NPM has been in use by Polly for a while now, the
majority of the tests continue using the LPM passes. This patch
ports the tests to use the NPM passes (for example, by replacing
a flag such as -polly-detect with -passes=polly-detect following
the NPM syntax for specifying passes) with some exceptions for
some missing features in the new passes. Additionally, the lit
substitution %loadPolly is replaced by the substitution of what
was %loadNPMPolly and %loadNPMPolly is removed.

Meinersbur commented on polly/test/Support/dumpfunction.ll:
```diff
@@ -1,9 +1,9 @@
 ; RUN: FileCheck --input-file=dumpfunction-caller-before.ll --check-prefix=CHECK --check-prefix=CALLER %s
 ;
-; RUN: opt %loadNPMPolly -O3 -polly -polly-position=before-vectorizer -polly-dump-after --disable-output %s
+; RUN: opt %loadPolly -O3 -polly -polly-position=before-vectorizer -polly-dump-after --disable-output %s
 ; RUN: FileCheck --input-file=dumpfunction-callee-after.ll --check-prefix=CHECK --check-prefix=CALLEE %s
 ; RUN: FileCheck --input-file=dumpfunction-caller-after.ll --check-prefix=CHECK --check-prefix=CALLER %s
```
These lines are identical to the ones above and can be removed. They were originally meant to test the legacy pass manager but the difference disappeared in 6c0fe14414cd0b3cbd96e809843ebbe3428da64e.

rahulana-quic replied:
These seem to be testing the polly dumps before and after and they both seem to be quite different. I think I have removed all the duplicate lines in tests that were testing both NPM and LPM passes previously (there were a few like this)

Meinersbur replied:
Correct, I seem to have looked for where the legacy pass tests are but git blame doesn't show them anymore (since remove). Thanks for clearing up.

Meinersbur reviewed (approved):
Thanks for all the work.
